perception/autoware_tensorrt_bevdet/src/bevdet_node.cpp

```diff
@@ -0,0 +1,261 @@
+    "~/input/topic_img_back_left/camera_info", rclcpp::SensorDataQoS{}.keep_last(1),
+    [this](const sensor_msgs::msg::CameraInfo::SharedPtr msg) { cameraInfoCallback(3, msg); });
+
+  sub_b_caminfo_ = this->create_subscription<sensor_msgs::msg::CameraInfo>(
+    "~/input/topic_img_back/camera_info", rclcpp::SensorDataQoS{}.keep_last(1),
+    [this](const sensor_msgs::msg::CameraInfo::SharedPtr msg) { cameraInfoCallback(4, msg); });
+
+  sub_br_caminfo_ = this->create_subscription<sensor_msgs::msg::CameraInfo>(
+    "~/input/topic_img_back_right/camera_info", rclcpp::SensorDataQoS{}.keep_last(1),
+    [this](const sensor_msgs::msg::CameraInfo::SharedPtr msg) { cameraInfoCallback(5, msg); });
+}
+
+void TRTBEVDetNode::callback(
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_fl_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_f_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_fr_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_bl_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_b_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_br_img)
+{
+  std::vector<cv::Mat> imgs;
+  auto clone_and_resize = [&](const sensor_msgs::msg::Image::ConstSharedPtr & msg) -> cv::Mat {
+    cv::Mat img = cv_bridge::toCvShare(msg, "bgr8")->image.clone();
+    if (img.size() != cv::Size(img_w_, img_h_)) {
+      cv::resize(img, img, cv::Size(img_w_, img_h_));  // img_w_ = 1600, img_h_ = 900
+    }
+    return img;
+  };
+
+  imgs.emplace_back(clone_and_resize(msg_fl_img));
+  imgs.emplace_back(clone_and_resize(msg_f_img));
+  imgs.emplace_back(clone_and_resize(msg_fr_img));
+  imgs.emplace_back(clone_and_resize(msg_bl_img));
+  imgs.emplace_back(clone_and_resize(msg_b_img));
+  imgs.emplace_back(clone_and_resize(msg_br_img));
+
+  imageTransport(imgs, imgs_dev_, img_w_, img_h_);
+
+  inference_input_.imgs_dev = imgs_dev_;
+
+  std::vector<Box> ego_boxes;
+  float inference_duration = 0.f;
+
+  bevdet_->DoInfer(inference_input_, ego_boxes, inference_duration);
+
+  autoware_perception_msgs::msg::DetectedObjects bevdet_objects;
+  bevdet_objects.header.frame_id = "base_link";
+  bevdet_objects.header.stamp = msg_f_img->header.stamp;
+
+  box3DToDetectedObjects(ego_boxes, bevdet_objects, class_names_, score_thre_, has_twist_);
+
+  pub_boxes_->publish(bevdet_objects);
+
+  if (debug_mode_) {
+    publishDebugMarkers(pub_markers_, bevdet_objects);
+  }
+}
+
+void TRTBEVDetNode::cameraInfoCallback(int idx, const sensor_msgs::msg::CameraInfo::SharedPtr msg)
+{
+  if (caminfo_received_[idx])
+    return;  // already received;  not expected to modify because of we init the model only once
+
+  if (!initialized_) {  // load image width and height from model config YAML
+    YAML::Node config = YAML::LoadFile(model_config_);
+    auto src_size = config["data_config"]["src_size"];
+    img_h_ = src_size[0].as<size_t>();  // height
+    img_w_ = src_size[1].as<size_t>();  // width
+    initialized_ = true;
+  }
```

Comment:
I think this part only runs once and the initialization does not require the camera info. So we can just move this into constructor.